addons/pos_restaurant/static/src/app/control_buttons/control_buttons.js

```diff
@@ -46,9 +46,23 @@ patch(ControlButtons.prototype, {
         });
     },
     clickTransferOrder() {
-        this.pos.orderToTransferUuid = this.pos.get_order().uuid;
+        this.dialog.closeAll();
+        this.pos.isOrderTransferMode = true;
+        const orderUuid = this.pos.get_order().uuid;
         this.pos.get_order().setBooked(true);
         this.pos.showScreen("FloorScreen");
+        document.addEventListener(
+            "click",
+            (ev) => {
+                this.pos.isOrderTransferMode = false;
+                const tableElement = ev.target.closest(".table");
+                if (!tableElement) {
+                    return;
+                }
+                this.pos.transferOrder(orderUuid, this.pos.getTableFromElement(tableElement));
+            },
+            { once: true }
+        );
```

Comment:
Following previous comment, its more easier to access the table record:
```suggestion
document.addEventListener(
    "click",
    (ev) => {
        this.pos.isOrderTransferMode = false;
        const tableElement = ev.target.closest(".table");
        const table = this.pos.getTableFromElement(tableElement);

        if (table) {
            this.pos.transferOrder(orderUuid, table);
        }
    },
    { once: true }
);
```